pootle/middleware/auth.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+"""
+Custom Pootle authentication middleware that takes care of returning Pootle's
+special `nobody` user instead of Django's own `AnonymousUser`.
+
+The code has been slightly adapted from django.contrib.auth.middleware.
+
+Note this customization would probably be unnecessary if there was a fix for
+https://code.djangoproject.com/ticket/20313
```

Comment:
This ticket being fixed also will render unnecessary `User.get` so perhaps it is a good idea to put some fixme note there as well pointing to this ticket.